lib/Borel/Types/Metric.hs

```diff
@@ -111,10 +113,32 @@ resourceMeasures = [Cumulative ..]
 
 prefixWeighting :: Prefix -> Double
 prefixWeighting Base = 1
-prefixWeighting Giga = 10^(9 :: Int)
-prefixWeighting Nano = 10^(-9 :: Int)
-prefixWeighting Mebi = 1024^(2 :: Int)
-prefixWeighting Mega = 10^(6 :: Int)
+prefixWeighting Giga = 10^^(9 :: Int)
+prefixWeighting Nano = 10^^(-9 :: Int)
+prefixWeighting Mebi = 1024^^(2 :: Int)
+prefixWeighting Mega = 10^^(6 :: Int)
+
+baseWeighting :: BaseUOM -> Double
+baseWeighting _ = 1
+
+weighting :: UOM -> Double
+weighting (UOM p b) = prefixWeighting p * baseWeighting b
+weighting (a `Times` b) = weighting a * weighting b
+
+extractBaseUOMs :: UOM -> MultiSet BaseUOM
+extractBaseUOMs (UOM _ b) = S.singleton b
+extractBaseUOMs (a `Times` b) = extractBaseUOMs a <> extractBaseUOMs b
+
+coercible :: UOM -> UOM -> Bool
+coercible x y = extractBaseUOMs x == extractBaseUOMs y
```

Comment:
If that is all we will ever use UOM for, and we know for a fact that the UOM units will always be commutative (nothing but times), and that will never change, then sure. It's not worth the extra effort to encode what is allowed as an UOM and how to convert between them.